### PR TITLE
More MySQL 8.x compatibility updates

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

- Update for MySQL 8.0 compatibility
- Update for Foreign Key support

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by dveeden** [comment]
Note that "MySQL 5.7 protocol" doesn't exist. MySQL has two protocol versions: v9 and v10. 

As [the v10 protocol is used since MySQL 3.21.0](https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_connection_phase_packets_protocol_handshake.html) the protocol version doesn't mean anything in most cases as there are many extensions based on [capability flags](https://dev.mysql.com/doc/dev/mysql-server/latest/group__group__cs__capabilities__flags.html). So best to say "MySQL Protocol" instead of "MySQL 5.7 Protocol" or "MySQL 8.0 Protocol".

File: tidb-cloud/tidb-cloud-faq.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ TiDB Cloud is a fully managed cloud Database as a Service of TiDB. It has an eas
 
 ### Is TiDB Cloud compatible with MySQL?
 
-Currently, TiDB Cloud supports the majority of MySQL 5.7 syntax with the exception of triggers, stored procedures, user-defined functions, and foreign keys. For more details, see [Compatibility with MySQL](https://docs.pingcap.com/tidb/stable/mysql-compatibility).
+Currently, TiDB Cloud supports the majority of MySQL 5.7 and MySQL 8.0 syntax with the exception of triggers, stored procedures, user-defined functions, and foreign keys. For more details, see [Compatibility with MySQL](https://docs.pingcap.com/tidb/stable/mysql-compatibility).
```
Comment (mjonss) [inline tidb-cloud/tidb-cloud-faq.md:28]: Is not foreign keys supported in TiDB Cloud?

**Review by mjonss — APPROVED** [review]
LGTM, only a question if foreign keys are not supported in TiDB Cloud? OK to fix in a separate issue/PR.

File: tidb-cloud/tidb-cloud-faq.md (hunk 1 above)
Comment (dveeden) [reply tidb-cloud/tidb-cloud-faq.md:28]:
```suggestion
Currently, TiDB Cloud supports the majority of MySQL 5.7 and MySQL 8.0 syntax with the exception of triggers, stored procedures, and user-defined functions. For more details, see [Compatibility with MySQL](/mysql-compatibility.md).
```

Good catch. I think that needs updating as well. See also https://docs.pingcap.com/tidbcloud/foreign-key

Looks like the link is also not correct.

**Review by bb7133 — APPROVED** [review]
LGTM

**Review by qiancai — APPROVED** [review]

**Pull request merged**